<?xml version="1.0" ?>
<launch>
    <include file="$(find test_pkg)/launch/ydlidar.launch" />
    <node name="qingzhou_bringup_node" pkg="test_pkg" type="qingzhou_bringup_node"/>

    <param name="mcuserialport" value="/dev/stm32board" type="string"  />
    <param name="mcubaudrate" value="115200" type="int" />
    <param name="calibrate_lineSpeed" value="0" type="int"  />
    <param name="calibrate_angularSpeed" value="0" type="int"  />
    <param name="ticksPerMeter" value="1191" type="int" />
    <param name="ticksPer2PI" value = "2145" type = "int"/>         
    <!-- 1224 -->
    <!-- 2100 -->
    
    <node pkg="imu_calib" type="apply_calib" name="apply_calib" output="screen" respawn="false">
      <param name="calib_file" value="$(find test_pkg)/calib/imu_calib.yaml" />
      <param name="calibrate_gyros" value="true" />
      <remap from="corrected" to="imu/data_raw" />
    </node>
      
    <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_madgwick" output="screen" respawn="true" >
      <param name="fixed_frame" value="odom" />
      <param name="use_mag" value="false" />
      <param name="publish_tf" value="false" />
      <param name="use_magnetic_field_msg" value="true" />  
      <param name="world_frame" value="enu" />
      <remap from="imu/data" to="imu_data" />
    </node>

     <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf"   respawn="true">
      <param name="output_frame" value="odom"/>
      <param name="base_footprint_frame" value="base_link"/>
      <param name="freq" value="30.0"/>
      <param name="sensor_timeout" value="1.0"/>
      <param name="odom_used" value="true"/>
      <param name="imu_used" value="true"/>
      <param name="vo_used" value="false"/>
      <param name="gps_used" value="false"/>
      <param name="debug" value="false"/>
      <param name="self_diagnose" value="false"/>
    </node>

    <node pkg="test_pkg" type="odom_ekf.py" name="odom_ekf">
      <remap from="input" to="robot_pose_ekf/odom_combined"/>
      <remap from="output" to="/odom_ekf"/>
    </node>  

    <!-- -0.08å’Œ0.05 -->
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_imulink"
    args="-0.08 0.0 0.05 0.0 0.0  0.0 /base_link /imu_link 50" />

</launch>